### ISSUE TITLE
feat(forge): new config option for filtering the contract artifacts which get generated

### ISSUE THREAD

**Issue by PaulRBerg** [foundry-rs/foundry#4979]
### Component

Forge

### Describe the feature you would like

The idea is to introduce a new config option called `filter` (or `sift`) that accepts a list of whitelisted contracts:

```toml
[profile.artifacts]
filter = [
    "Foo",
    "Bar",
]
```

When it sees this config, Forge generates artifacts only for the contracts listed under `filter`, regardless of whatever other contracts may be present in the repository.

The primary use case is shipping a Node.js package that should include only the production contracts' artifacts - though there may be more use cases than just that. Anything related to artifact generation may benefit from such a feature.

### Additional context

In our project, we wrote a [Bash script](https://gist.github.com/PaulRBerg/6cdca9662bb51474b2580fddf4c03c0c) because we couldn't find a way to filter contracts with Foundry itself.

**Comment by PaulRBerg** [comment]
Sharing the Bash scripts we have used to prepare the Sablier V2 artifacts:

- https://github.com/sablier-labs/v2-core/blob/431f79c248bbb025f5581141cc605f220a0fb404/shell/prepare-artifacts.sh
- https://github.com/sablier-labs/v2-periphery/blob/747fc5e789fc17b2a3c1560bf70b2bbcb597901f/shell/prepare-artifacts.sh

We would love to be able to delete the scripts 😁

**Comment by grandizzy** [comment]
@PaulRBerg we have a PR in works https://github.com/foundry-rs/foundry/pull/8668 would this solve your request here too? thank you!

**Comment by PaulRBerg** [comment]
Thanks @grandizzy, tagging @smol-ninja and @andreivladbrg for taking a look.

**Comment by smol-ninja** [comment]
@grandizzy as per [this comment](https://github.com/foundry-rs/foundry/pull/8668#issuecomment-2408566530), that PR does not seem to solve this issue. 

However, there is a mention of `--skip` flag that can be used to dictate artifacts. Is that something already a part of foundry (couldn't find it in the book though)?

**Comment by grandizzy** [comment]
> However, there is a mention of `--skip` flag that can be used to dictate artifacts. Is that something already a part of foundry (couldn't find it in the book though)?

nope, nothing yet, just wanted to check if the PR could work for you too. thank you
